android/library/src/main/java/com/polidea/multiplatformbleadapter/BleModule.java

```diff
@@ -157,533 +127,793 @@ public void destroyClient() {
-                        String deviceId = scanResult.getBleDevice().getMacAddress();
-                        if (!discoveredDevices.containsKey(deviceId)) {
-                            discoveredDevices.put(deviceId, new Device(scanResult.getBleDevice(), null));
-                        }
-                        sendEvent(Event.ScanEvent, scanConverter.toJSCallback(scanResult));
-                    }
-                }, new Action1<Throwable>() {
-                    @Override
-                    public void call(Throwable throwable) {
-                        sendEvent(Event.ScanEvent, errorConverter.toError(throwable).toJSCallback());
-                    }
-                });
+        safeConnectToDevice(
+                device,
+                connectionOptions.getAutoConnect(),
+                connectionOptions.getRequestMTU(),
+                connectionOptions.getRefreshGattMoment(),
+                connectionOptions.getTimeoutInMillis(),
+                connectionOptions.getConnectionPriority(),
+                onSuccessCallback, onErrorCallback);
     }
 
-    @ReactMethod
-    public void stopDeviceScan() {
-        if (scanSubscription != null) {
-            if (!scanSubscription.isUnsubscribed()) {
-                scanSubscription.unsubscribe();
+    @Override
+    public void cancelDeviceConnection(String deviceIdentifier,
+                                       OnSuccessCallback<Device> onSuccessCallback,
+                                       OnErrorCallback onErrorCallback) {
+        if (rxBleClient == null) {
+            throw new IllegalStateException("BleManager not created when tried cancel device connection");
+        }
+
+        final RxBleDevice device = rxBleClient.getBleDevice(deviceIdentifier);
+
+        if (connectingDevices.removeSubscription(deviceIdentifier) && device != null) {
+            onSuccessCallback.onSuccess(rxBleDeviceToDeviceMapper.map(device));
+        } else {
+            if (device == null) {
+                onErrorCallback.onError(BleErrorUtils.deviceNotFound(deviceIdentifier));
+            } else {
+                onErrorCallback.onError(BleErrorUtils.deviceNotConnected(deviceIdentifier));
             }
-            scanSubscription = null;
         }
     }
 
-    // Mark: Device management ---------------------------------------------------------------------
-
-    @ReactMethod
-    public void devices(final ReadableArray deviceIdentifiers, final Promise promise) {
+    @Override
+    public void isDeviceConnected(String deviceIdentifier,
+                                  OnSuccessCallback<Boolean> onSuccessCallback,
+                                  OnErrorCallback onErrorCallback) {
         if (rxBleClient == null) {
-            throw new IllegalStateException("BleManager not created when tried connecting to device");
+            throw new IllegalStateException("BleManager not created when tried cancel device connection");
```

Comment:
I'm not really sure what this message is supposed to mean.

Reply:
And why does message says something about BleManager when it checks whether rxBleClient exists?